src/librekey/key_store_g10.cpp

```diff
@@ -814,7 +814,8 @@ parse_protected_seckey(pgp_key_pkt_t *seckey, s_exp_t *s_exp, const char *passwo
         goto done;
     }
     sub_el = sub_element_at(alg, 0);
-    if (strncmp("sha1", (const char *) sub_el->block.bytes, sub_el->block.len) != 0) {
+    if ((sub_el->block.len != 4) ||
```

Comment:
If the length is fixed,
it would be more evident to write
`(sub_el->block.len != 4) || memcmp("sha1", sub_el->block.bytes, 4)`
Agree?

Reply:
@rrrooommmaaa Agree, thanks for pointing to this.